### PR TITLE
Add active game to top of page

### LINKED ISSUE #2: Increase size of today's match
More prominently display today's match for easier viewing.

### PR DESCRIPTION
**Pull request by davwheat**
image

Closes #2

### CODE REVIEW

**Comment by davwheat** [comment]
Ready for review!

**Comment by Gum-Joe** [comment]
```diff
diff --git a/src/components/GameGrid/Rounds/GameRound.tsx b/src/components/GameGrid/Rounds/GameRound.tsx
index 0dbbc54..e91603f 100644
--- a/src/components/GameGrid/Rounds/GameRound.tsx
+++ b/src/components/GameGrid/Rounds/GameRound.tsx
@@ -11,9 +11,10 @@ import FormatDate from '../../../functions/formatDate'
 import clsx from 'clsx'
 
 interface Props {
-  data: SinglePoll
-  noDate: boolean
-  large: boolean
+  data: SinglePoll;
+  noDate: boolean;
+  /** Show a large graph. Used for the active game box. */
+  large: boolean;
 }
 
 const useStyles = makeStyles({
@@ -36,6 +37,9 @@ const useStyles = makeStyles({
   },
 })
 
+/**
+ * Provides a way to show each game, and the corresponding graph
+ */
 export default function GameRound(props: Props) {
   const { data, noDate, large } = props
   const classes = useStyles()

```

**Comment by Gum-Joe** [comment]
```diff
diff --git a/src/components/GameGrid/Rounds/GameRound.tsx b/src/components/GameGrid/Rounds/GameRound.tsx
index 0dbbc54..1912294 100644
--- a/src/components/GameGrid/Rounds/GameRound.tsx
+++ b/src/components/GameGrid/Rounds/GameRound.tsx
@@ -11,9 +11,10 @@ import FormatDate from '../../../functions/formatDate'
 import clsx from 'clsx'
 
 interface Props {
-  data: SinglePoll
-  noDate: boolean
-  large: boolean
+  data: SinglePoll;
+  noDate: boolean;
+  /** Show a large graph. Used for the active game box. */
+  large: boolean;
 }
 
 const useStyles = makeStyles({
@@ -36,9 +37,12 @@ const useStyles = makeStyles({
   },
 })
 
+/**
+ * Provides a way to show each game, and the corresponding graph
+ */
 export default function GameRound(props: Props) {
-  const { data, noDate, large } = props
-  const classes = useStyles()
+  const { data, noDate, large } = props;
+  const classes = useStyles();
 
   const teamInfo = {
     team1: GetTocName(data.votesInfo[0].tocReportingMark),
diff --git a/src/components/GameGrid/Rounds/KnockoutRounds.tsx b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
index 4569719..84c574a 100644
--- a/src/components/GameGrid/Rounds/KnockoutRounds.tsx
+++ b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
@@ -34,15 +34,14 @@ const useStyles = makeStyles({
   },
 })
 
-interface KnockoutRoundData {
-  [key: number]: SinglePoll
-}
+/** Basically equivalent to an interface with { [key: number]: SinglePoll } */
+type KnockoutRoundData = Record<number, SinglePoll>;
 
 interface Props {
   knockoutRoundData: KnockoutRoundData
 }
 
-export default function KnockoutRounds(props: Props) {
+const KnockoutRounds: React.FunctionComponent<Props> = (props: Props) => {
   const { knockoutRoundData } = props
   const classes = useStyles()
 
@@ -75,3 +74,5 @@ export default function KnockoutRounds(props: Props) {
     </>
   )
 }
+
+export default KnockoutRounds;

```

**Comment by Gum-Joe** [comment]
```diff
diff --git a/src/components/GameGrid/Rounds/GameRound.tsx b/src/components/GameGrid/Rounds/GameRound.tsx
index 3d91e74..8019377 100644
--- a/src/components/GameGrid/Rounds/GameRound.tsx
+++ b/src/components/GameGrid/Rounds/GameRound.tsx
@@ -51,7 +51,7 @@ const useStyles = makeStyles({
  * @param {Props} props
  * @return {React.ReactNode}
  */
-export default function GameRound({ data, noDate, large }: Props) {
+const GameRound: React.FunctionComponent<Props> = ({ data, noDate, large }: Props) => {
   const classes = useStyles()
 
   const teamInfo = {
@@ -90,3 +90,5 @@ export default function GameRound({ data, noDate, large }: Props) {
     </div>
   )
 }
+
+export default GameRound;
\ No newline at end of file
diff --git a/src/components/GameGrid/Rounds/KnockoutRounds.tsx b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
index 4569719..84c574a 100644
--- a/src/components/GameGrid/Rounds/KnockoutRounds.tsx
+++ b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
@@ -34,15 +34,14 @@ const useStyles = makeStyles({
   },
 })
 
-interface KnockoutRoundData {
-  [key: number]: SinglePoll
-}
+/** Basically equivalent to an interface with { [key: number]: SinglePoll } */
+type KnockoutRoundData = Record<number, SinglePoll>;
 
 interface Props {
   knockoutRoundData: KnockoutRoundData
 }
 
-export default function KnockoutRounds(props: Props) {
+const KnockoutRounds: React.FunctionComponent<Props> = (props: Props) => {
   const { knockoutRoundData } = props
   const classes = useStyles()
 
@@ -75,3 +74,5 @@ export default function KnockoutRounds(props: Props) {
     </>
   )
 }
+
+export default KnockoutRounds;
diff --git a/src/typography/loud.tsx b/src/typography/loud.tsx
index f0dc6d4..55796d7 100644
--- a/src/typography/loud.tsx
+++ b/src/typography/loud.tsx
@@ -26,22 +26,22 @@ const useStyles = makeStyles({
 })
 
 interface Props {
-  children: React.ReactNode
+  children: React.ReactNode;
   /**
    * Extra classes to add in addition to those supplied by this component
    * @default undefined
    */
-  className?: string
+  className?: string;
   /**
    * Should remove top and bottom padding
    * @default false
    */
-  noPadding?: boolean
+  noPadding?: boolean;
   /**
    * Should paragraph be centred
    * @default false
    */
-  center?: boolean
+  center?: boolean;
 }
 
 export default function Loud({ children, className, noPadding, center }: Props) {

```

**Comment by Gum-Joe** [comment]
```diff
diff --git a/src/components/GameGrid/Rounds/GameRound.tsx b/src/components/GameGrid/Rounds/GameRound.tsx
index 3d91e74..8019377 100644
--- a/src/components/GameGrid/Rounds/GameRound.tsx
+++ b/src/components/GameGrid/Rounds/GameRound.tsx
@@ -51,7 +51,7 @@ const useStyles = makeStyles({
  * @param {Props} props
  * @return {React.ReactNode}
  */
-export default function GameRound({ data, noDate, large }: Props) {
+const GameRound: React.FunctionComponent<Props> = ({ data, noDate, large }: Props) => {
   const classes = useStyles()
 
   const teamInfo = {
@@ -90,3 +90,5 @@ export default function GameRound({ data, noDate, large }: Props) {
     </div>
   )
 }
+
+export default GameRound;
\ No newline at end of file
diff --git a/src/components/GameGrid/Rounds/KnockoutRounds.tsx b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
index 4569719..84c574a 100644
--- a/src/components/GameGrid/Rounds/KnockoutRounds.tsx
+++ b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
@@ -34,15 +34,14 @@ const useStyles = makeStyles({
   },
 })
 
-interface KnockoutRoundData {
-  [key: number]: SinglePoll
-}
+/** Basically equivalent to an interface with { [key: number]: SinglePoll } */
+type KnockoutRoundData = Record<number, SinglePoll>;
 
 interface Props {
   knockoutRoundData: KnockoutRoundData
 }
 
-export default function KnockoutRounds(props: Props) {
+const KnockoutRounds: React.FunctionComponent<Props> = (props: Props) => {
   const { knockoutRoundData } = props
   const classes = useStyles()
 
@@ -75,3 +74,5 @@ export default function KnockoutRounds(props: Props) {
     </>
   )
 }
+
+export default KnockoutRounds;
diff --git a/src/typography/loud.tsx b/src/typography/loud.tsx
index f0dc6d4..55796d7 100644
--- a/src/typography/loud.tsx
+++ b/src/typography/loud.tsx
@@ -26,22 +26,22 @@ const useStyles = makeStyles({
 })
 
 interface Props {
-  children: React.ReactNode
+  children: React.ReactNode;
   /**
    * Extra classes to add in addition to those supplied by this component
    * @default undefined
    */
-  className?: string
+  className?: string;
   /**
    * Should remove top and bottom padding
    * @default false
    */
-  noPadding?: boolean
+  noPadding?: boolean;
   /**
    * Should paragraph be centred
    * @default false
    */
-  center?: boolean
+  center?: boolean;
 }
 
 export default function Loud({ children, className, noPadding, center }: Props) {

```

File: src/components/GameGrid/Rounds/KnockoutRounds.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+interface KnockoutRoundData {
+  [key: number]: SinglePoll
+}
+
+interface Props {
+  knockoutRoundData: KnockoutRoundData
+}
+
+export default function KnockoutRounds(props: Props) {
+  const { knockoutRoundData } = props
```
Comment (Gum-Joe) [inline src/components/GameGrid/Rounds/KnockoutRounds.tsx:46]: Overall, look great. Just some minor changes (mostly types).
 
Some major changes I'd like to suggest to the site before I give approval:
1. Move the Actibe Game above the KnockoutRound header
2. Make the active game graph a tad taller.

Other than that, look great, graphs are really nice and the styling on the active game is great.  TSX transition has also gone well (just please remember semi-colons)

File: src/components/GameGrid/Rounds/KnockoutRounds.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const classes = useStyles()
+
```
Comment (Gum-Joe) [inline src/components/GameGrid/Rounds/KnockoutRounds.tsx:48]: End review comment.

**Comment by Gum-Joe** [comment]
```diff
diff --git a/src/components/GameGrid/Rounds/GameRound.tsx b/src/components/GameGrid/Rounds/GameRound.tsx
index 3d91e74..8019377 100644
--- a/src/components/GameGrid/Rounds/GameRound.tsx
+++ b/src/components/GameGrid/Rounds/GameRound.tsx
@@ -51,7 +51,7 @@ const useStyles = makeStyles({
  * @param {Props} props
  * @return {React.ReactNode}
  */
-export default function GameRound({ data, noDate, large }: Props) {
+const GameRound: React.FunctionComponent<Props> = ({ data, noDate, large }: Props) => {
   const classes = useStyles()
 
   const teamInfo = {
@@ -90,3 +90,5 @@ export default function GameRound({ data, noDate, large }: Props) {
     </div>
   )
 }
+
+export default GameRound;
\ No newline at end of file
diff --git a/src/components/GameGrid/Rounds/KnockoutRounds.tsx b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
index 4569719..84c574a 100644
--- a/src/components/GameGrid/Rounds/KnockoutRounds.tsx
+++ b/src/components/GameGrid/Rounds/KnockoutRounds.tsx
@@ -34,15 +34,14 @@ const useStyles = makeStyles({
   },
 })
 
-interface KnockoutRoundData {
-  [key: number]: SinglePoll
-}
+/** Basically equivalent to an interface with { [key: number]: SinglePoll } */
+type KnockoutRoundData = Record<number, SinglePoll>;
 
 interface Props {
   knockoutRoundData: KnockoutRoundData
 }
 
-export default function KnockoutRounds(props: Props) {
+const KnockoutRounds: React.FunctionComponent<Props> = (props: Props) => {
   const { knockoutRoundData } = props
   const classes = useStyles()
 
@@ -75,3 +74,5 @@ export default function KnockoutRounds(props: Props) {
     </>
   )
 }
+
+export default KnockoutRounds;
diff --git a/src/typography/loud.tsx b/src/typography/loud.tsx
index f0dc6d4..55796d7 100644
--- a/src/typography/loud.tsx
+++ b/src/typography/loud.tsx
@@ -26,22 +26,22 @@ const useStyles = makeStyles({
 })
 
 interface Props {
-  children: React.ReactNode
+  children: React.ReactNode;
   /**
    * Extra classes to add in addition to those supplied by this component
    * @default undefined
    */
-  className?: string
+  className?: string;
   /**
    * Should remove top and bottom padding
    * @default false
    */
-  noPadding?: boolean
+  noPadding?: boolean;
   /**
    * Should paragraph be centred
    * @default false
    */
-  center?: boolean
+  center?: boolean;
 }
 
 export default function Loud({ children, className, noPadding, center }: Props) {

```

**Review by Gum-Joe — CHANGES_REQUESTED** [review]
Overall, look great. Just some minor changes (mostly types). See above for comment.

Some major changes I'd like to suggest to the site before I give approval:

2. Move the Actibe Game above the KnockoutRound header
3. Make the active game graph a tad taller.

Other than that, look great, graphs are really nice and the styling on the active game is great. TSX transition has also gone well (just please remember semi-colons).

File: src/components/GameGrid/Rounds/KnockoutRounds.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+interface KnockoutRoundData {
+  [key: number]: SinglePoll
+}
+
+interface Props {
+  knockoutRoundData: KnockoutRoundData
+}
+
+export default function KnockoutRounds(props: Props) {
```
[the remark this answers is not in the document]
Comment (Gum-Joe) [reply src/components/GameGrid/Rounds/KnockoutRounds.tsx]: Have added this.

File: src/components/GameGrid/Rounds/KnockoutRounds.tsx (hunk 1 above)
Comment (Gum-Joe) [reply src/components/GameGrid/Rounds/KnockoutRounds.tsx:45]: Have made type changes because I was unable to suggest them.

**Comment by davwheat** [comment]
image

**Review by Gum-Joe — APPROVED** [review]
Summary: Approved, but there's some issues on mobile.

Ok so desktop wise, the site's approved! It looks great. There's some issues on mobile (see below), but I decided the scope of fixing those falls outside this PR.

Code wise, look great, all I would say is eventually (in a future PR) the `models` should be migrated to TS.  They also look really well documented - well done!

Mobile wise, there's some issues:
On an iPhone 6/7/8. the site has overflow:
image

On iPhone SE gen 1, not much better:
image

Landscape though, both look perfectly fine:
image

What might be worth doing is to change the notice to "best viewed on a larger device or in landscape".

(tip: use the device emulation feature in dev tools to view the site on mobile in different size, etc)

This issue is not related to moveing the active game to the top of the page, hence I decided they are out of the scope of this PR and so don't warrant a "Request for changes".

Since everything else is brilliant, I therefore APPROVE this PR.

**Pull request merged**